### PR TITLE
Smoke test for index implementations

### PR DESCRIPTION
**Pull request by niklasb**
added naive BRUTE_IDX index implementation and a python script to test indexes against each other.

### CODE REVIEW

File: include/surf/config.hpp [hunk 1]
```diff
@@ -60,6 +60,8 @@ const std::string KEY_WEIGHTS = "weights";
 const std::string KEY_WEIGHTS_G = "weights_g";
 const std::string KEY_W_AND_P = "W_and_P";
 const std::string KEY_W_AND_P_G = "W_and_P_g";
+const std::string KEY_DOC_SPLITTERS = "doc_splitters";
```
Comment (jlabeit) [inline include/surf/config.hpp]: Looks like this is the same as KEY_DOCBORDER and KEY_DOCBORDER_RANK.
Maybe you can just use them.

Comment (niklasb) [reply include/surf/config.hpp]: Done

File: include/surf/idx_brute.hpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    size_type serialize(std::ostream& out,
+                        structure_tree_node* v = nullptr,
+                        std::string name = "") const {
+        sdsl::structure_tree_node* child =
+            sdsl::structure_tree::add_child(v, name, util::class_name(*this));
+        size_type written_bytes = 0;
+        written_bytes += m_csa.serialize(out, child, "CSA");
+        written_bytes += m_doc_splitters.serialize(out, child, "DOC_SPLITTERS");
```
Comment (jlabeit) [inline include/surf/idx_brute.hpp]: "DOC_SPLITTERS" -> "BORDER"
"DOC_SPLITTERS_RANK" -> "BORDER_RANK"

Comment (niklasb) [reply include/surf/idx_brute.hpp]: Already fixed no?

File: tools/compare.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            f.flush()
+            for config in args.targets:
+                print '    Running with %s' % config
+                out = subprocess.check_output([
+                    'build/surf_query-%s' % config,
+                    '-c', args.c,
+                    '-q', f.name,
+                    '-v'
+                ])
+                lines = out.strip().splitlines()
```
Comment (jlabeit) [inline tools/compare.py:89]: I think it makes sense to add some logic here to also check if the document ids are correct.

Comment (niklasb) [reply tools/compare.py:89]: Done

**Pull request merged**